src/App.tsx

```diff
@@ -1,61 +1,157 @@
+import React, {
+  ChangeEvent, FormEvent, useState,
+} from 'react';
+import fetchedUsers from './api/users';
+import fetchedTodos from './api/todos';
+import { User } from './types';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todosWithAssociatedUsers = fetchedTodos.map(todo => ({
+  ...todo,
+  user: fetchedUsers.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todoItems, setTodoItems] = useState(todosWithAssociatedUsers);
+  const [formValues, setFormValues] = useState({ title: '', user: '' });
+  const [titleValidation, setTitleValidation] = useState({
+    value: '',
+    isValid: false,
+    displayError: false,
+  });
+
+  const [userValidation, setUserValidation] = useState({
+    value: '',
+    isValid: false,
+    displayError: false,
+  });
```

Comment:
that is okay for now, but if you want to store complex object as state, then read about [useReducer](https://react.dev/reference/react/useReducer#examples-basic) or watch a [video](https://www.youtube.com/watch?v=kK_Wqx3RnHk)